src/sgame/sg_spawn.cpp

```diff
@@ -790,6 +790,73 @@ static void G_ParseField( const char *key, const char *rawString, gentity_t *ent
 		G_WarnAboutDeprecatedEntityField(entity, fieldDescriptor->replacement, key, fieldDescriptor->versionState );
 }
 
+static std::unordered_map< std::string, int > autoIdCounters;
+static const std::set< std::string > autoClassnames = {
+	"env_afx_ammo", "env_afx_gravity", "env_afx_heal", "env_afx_hurt",
```

Comment:
Why do you need a closed list instead of just doing it for every class?

Reply:
If you have `std::map<std::string, int> m` you can do `n = ++m[classname]` and the map entry will be automatically created and initialized to 0 if it doesn't exist.

Reply:
Thank you. That was a relic from a more naive method.